sdks/go/pkg/beam/runners/prism/internal/engine/teststream.go

```diff
@@ -142,10 +142,8 @@ func (ts *testStreamHandler) UpdateHold(em *ElementManager, newHold mtime.Time)
 	ts.currentHold = newHold
 	ss.watermarkHolds.Add(ts.currentHold, 1)
 
-	// kick the TestStream and Impulse stages too.
+	// kick the TestStream stage to ensure downstream watermark propagation.
```

Comment:
Non-blocking question - What does 'kick' mean in this context?

Reply:
'Kick' signals the element manager to integrate the intent of the testStreamHandler's watermark settings under test with related stages' watermark. In other words, if there are remaining elements in test stream, we prevent prematurely setting all the stages' output watermark to infinity. The underlying mechanism by which we signal i.e. 'kick' is to merge the testStreamHandler's ID to the element manager's changedStages property.

Reply:
That's right.

Like when you kick a ball, or engage in percussive maintenance on any mechanical device, this "kick" gets the Element Manager moving again, by saying a stage needs to be evaluated, and as it says, to ensure downstream watermark propagation.